src/links.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import {
+  Link as ReactRouterLink,
+  NavLink as ReactRouterNavLink,
+} from 'react-router-dom'
+
+function isRoutingUrl (to) {
+  if (typeof to !== 'string') return true
+  return !to.match(/^#/) && !to.match(/^[a-z]{1,10}:\/\//)
```

Comment:
Will this work when using the [`hash` version of the Router](https://github.com/nozzle/react-static#router)?

Reply:
Good point, I haven’t tried it.

It should work, as the user of the components should still provide a proper, non-fragment URL, it is the responsibility of `history`/`react-router` to encode the output in whatever format they currently need.

Reply:
I can try and verify it, maybe tomorrow.

Reply:
No meaningful BC breakages, here are the findings:

1. `<Router type="hash">` and regular links keep working. `<Link to="about">` matches our `isRoutingUrl` check and thus gets passed through to react-router, which renders a `<a href="#/about">`.

2. `<Router type="hash">` and links that manually have the hash route kind of work. `<Link to="#/about">` does not match our `isRoutingUrl` check and gets rendered with the newly introduced shortcut into a `<a href="#/about">`. react-router handles this links, so it responds to user interaction and changes routes, but it does not have `aria-current`. We could attempt to distinguish `to="#/about"` as a non-fragment link, but technically ids [can start with a slash](https://www.w3.org/TR/2011/WD-html5-20110525/elements.html#the-id-attribute). The usage `<Link to="#/about">` is already kind of wrong, and with proper history support [nearly universal](https://caniuse.com/#feat=history), I don’t think we have to worry about this.

3. `<Router type="hash">` and fragment links are still broken after the latest changes. `<Link to="#subheading">` gets rendered with the newly introduced shortcut into a `<a href="#subheading">`, but is still handled by react-router to navigate to `#/subheading`, ignoring the fragment link. This [was already broken](https://github.com/ReactTraining/react-router/issues/394) with the react-router `<Link>`, and is still broken with our manually rendered `<a>`. Nothing to see here.